widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.kt

```diff
@@ -0,0 +1,186 @@
+
+    private val defaultCallConfiguration: CallConfiguration
+        get() = CallConfiguration(engagementConfiguration)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        this.enableEdgeToEdge()
+        super.onCreate(savedInstanceState)
+        Logger.i(TAG, "Create Chat screen")
+        setContentView(R.layout.chat_activity)
+        chatView = findViewById(R.id.chat_view)
+
+        onBackPressedDispatcher.addCallback(this, object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() {
+                chatView.onBackPressed()
+            }
+        })
+
+        // Legacy company name support
+        sdkConfigurationManager.setLegacyCompanyName(intent.getStringExtra(GliaWidgets.COMPANY_NAME))
+
+        chatView.setOnTitleUpdatedListener(this::setTitle)
+        engagementConfiguration = createEngagementConfiguration(intent)
+        if (intent.hasExtra(GliaWidgets.USE_OVERLAY)) {
+            // Integrator has passed a deprecated GliaWidgets.USE_OVERLAY parameter with Intent
+            // Override bubble configuration with USE_OVERLAY value
+            val useOverlay = intent.getBooleanExtra(GliaWidgets.USE_OVERLAY, true)
+            sdkConfigurationManager.setLegacyUseOverlay(useOverlay)
+        }
+
+        if (!chatView.shouldShow()) {
+            finishAndRemoveTask()
+            return
+        }
+
+        chatView.setConfiguration(engagementConfiguration)
+        chatView.setUiTheme(engagementConfiguration.runTimeTheme)
+        chatView.setOnBackClickedListener(::finish)
+        chatView.setOnBackToCallListener(::backToCallScreen)
+
+        // In case the engagement ends, Activity is removed from the device's Recents menu
+        // to avoid app users to accidentally start queueing for another call when they resume
+        // the app from the Recents menu and the app's backstack was empty.
+        chatView.setOnEndListener(::finishAndRemoveTask)
+
+        chatView.setOnMinimizeListener(::finish)
+        chatView.setOnNavigateToCallListener(this::startCallScreen)
+        chatView.startChat(
+            engagementConfiguration.companyName,
+            engagementConfiguration.queueIds,
+            engagementConfiguration.contextAssetId,
+            engagementConfiguration.screenSharingMode,
+            engagementConfiguration.chatType ?: ChatType.LIVE_CHAT
+        )
+    }
+
+    override fun onResume() {
+        chatView.onResume()
+        super.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        chatView.onPause()
+    }
+
+    override fun onDestroy() {
+        chatView.onDestroyView()
+        Logger.i(TAG, "Destroy Chat screen")
+        super.onDestroy()
+    }
+
+    private fun createEngagementConfiguration(intent: Intent): EngagementConfiguration = EngagementConfiguration(intent)
+
+    //TODO: Check why theme attribute is not anymore used
```

Comment:
Is it a leftover?

Reply:
No, it is more for the next tickets in this story.